design/interactable.behavior.md

```diff
@@ -0,0 +1,39 @@
+#  Interactable Behavior
+
+The interactable behavior allows the user to assign this behavior to an actor so that the actor can be held and
+used.
+
+## Architecture
+
+Create a new behavior that is an `InteractableBehavior` for being able to add to an actor.
+
+``` ts
+/**
+ * Interactable behavior class containing the target behavior actions.
+ */
+export class InteractableBehavior extends TargetBehavior {
```

Comment:
How would the app be responsive to state changes (when holding or using changes)? ButtonBehavior has events/callbacks for onButton, onClick, etc. Would it make sense to add callbacks for onHeld, onDropped, etc? Or can this be done from DiscreteAction?

Reply:
I didn't show that API, but we provide helper functions for a named API to those discrete actions, but those state changes to come through the actor.  I will add the API for the helper functions like the other behaviors to the design for these.  It was redundant so didn't think it was necessary, but can see how it would be helpful in the design doc.